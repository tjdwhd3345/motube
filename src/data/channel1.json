{
  "url": "https://youtube.googleapis.com/youtube/v3/channels?key=AIzaSyDMH41LEzdH2pjqKYZkIIgCg4dpMJtcDAI&maxResults=30&regionCode=KR&part=snippet&id=UCDNvRZRgvkBTUkQzFoT_8rA",
  "kind": "youtube#channelListResponse",
  "etag": "1UtySItUSQd5fogrx1t3yHC-JLI",
  "pageInfo": {
    "totalResults": 1,
    "resultsPerPage": 30
  },
  "items": [
    {
      "kind": "youtube#channel",
      "etag": "6-8daZbrZu_zg-5PiNgHhj6TiaA",
      "id": "UCDNvRZRgvkBTUkQzFoT_8rA",
      "snippet": {
        "title": "뜬뜬 DdeunDdeun",
        "description": "\n좋은 사람, 좋은 이야기, 좋은 웃음\n우리 '뜬뜬'한 사이 돼봐요💕 \n\n🤓유선배 복지 콘텐츠🗣토크찌개맛집🥘\n🐓핑계고🐓는 불시에 찾아옵니다! \n----------------------------------------\n🥂그저 술 마시며 게임하는 고품격 사교모임🥂\n🎲빰빰소셜클럽🎲금요일 밤 10시 OPEN🔔\n\n🙏🏻 plz 구독☝알림설정🔔 plz 🙏🏻\n\n뭐든지 문의할 땐, ddeunddeun@antenna.co.kr\n인스타 놀러올 땐, @ddeun._.ddeun\n",
        "customUrl": "@ddeunddeun",
        "publishedAt": "2022-10-04T07:24:23.973864Z",
        "thumbnails": {
          "default": {
            "url": "https://yt3.ggpht.com/aOgjye3sMIxNl2SW2wAQZpZWUXzZ5Rg0rNITacRQKVfXvF9cnPWb77G3_gH5s2Zyw241BXWYWg=s88-c-k-c0x00ffffff-no-rj",
            "width": 88,
            "height": 88
          },
          "medium": {
            "url": "https://yt3.ggpht.com/aOgjye3sMIxNl2SW2wAQZpZWUXzZ5Rg0rNITacRQKVfXvF9cnPWb77G3_gH5s2Zyw241BXWYWg=s240-c-k-c0x00ffffff-no-rj",
            "width": 240,
            "height": 240
          },
          "high": {
            "url": "https://yt3.ggpht.com/aOgjye3sMIxNl2SW2wAQZpZWUXzZ5Rg0rNITacRQKVfXvF9cnPWb77G3_gH5s2Zyw241BXWYWg=s800-c-k-c0x00ffffff-no-rj",
            "width": 800,
            "height": 800
          }
        },
        "localized": {
          "title": "뜬뜬 DdeunDdeun",
          "description": "\n좋은 사람, 좋은 이야기, 좋은 웃음\n우리 '뜬뜬'한 사이 돼봐요💕 \n\n🤓유선배 복지 콘텐츠🗣토크찌개맛집🥘\n🐓핑계고🐓는 불시에 찾아옵니다! \n----------------------------------------\n🥂그저 술 마시며 게임하는 고품격 사교모임🥂\n🎲빰빰소셜클럽🎲금요일 밤 10시 OPEN🔔\n\n🙏🏻 plz 구독☝알림설정🔔 plz 🙏🏻\n\n뭐든지 문의할 땐, ddeunddeun@antenna.co.kr\n인스타 놀러올 땐, @ddeun._.ddeun\n"
        },
        "country": "KR"
      }
    }
  ]
}
