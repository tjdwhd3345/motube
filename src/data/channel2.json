{
  "url": "https://youtube.googleapis.com/youtube/v3/channels?key=AIzaSyDMH41LEzdH2pjqKYZkIIgCg4dpMJtcDAI&maxResults=30&regionCode=KR&part=snippet&id=UCyn-K7rZLXjGl7VXGweIlcA",
  "kind": "youtube#channelListResponse",
  "etag": "GUFYtCWED-_IGobdmQqA7pvQLA4",
  "pageInfo": {
    "totalResults": 1,
    "resultsPerPage": 30
  },
  "items": [
    {
      "kind": "youtube#channel",
      "etag": "9q2OSWT0-I8hqcvNS5Yc3nSnWug",
      "id": "UCyn-K7rZLXjGl7VXGweIlcA",
      "snippet": {
        "title": "백종원 PAIK JONG WON",
        "description": "",
        "customUrl": "@paik_jongwon",
        "publishedAt": "2018-03-05T02:51:11Z",
        "thumbnails": {
          "default": {
            "url": "https://yt3.ggpht.com/J3OI66Bc7T3nheyKJKAkXR6tb-_bpCsoRMTFoslOBCXI3DpVY8eFY4LZWzww3BEgkRjMOEoI=s88-c-k-c0x00ffffff-no-rj",
            "width": 88,
            "height": 88
          },
          "medium": {
            "url": "https://yt3.ggpht.com/J3OI66Bc7T3nheyKJKAkXR6tb-_bpCsoRMTFoslOBCXI3DpVY8eFY4LZWzww3BEgkRjMOEoI=s240-c-k-c0x00ffffff-no-rj",
            "width": 240,
            "height": 240
          },
          "high": {
            "url": "https://yt3.ggpht.com/J3OI66Bc7T3nheyKJKAkXR6tb-_bpCsoRMTFoslOBCXI3DpVY8eFY4LZWzww3BEgkRjMOEoI=s800-c-k-c0x00ffffff-no-rj",
            "width": 800,
            "height": 800
          }
        },
        "localized": {
          "title": "백종원 PAIK JONG WON",
          "description": ""
        },
        "country": "KR"
      }
    }
  ]
}
