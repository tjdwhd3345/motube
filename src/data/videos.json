{
  "url": "https://youtube.googleapis.com/youtube/v3/videos?key=AIzaSyDMH41LEzdH2pjqKYZkIIgCg4dpMJtcDAI&maxResults=30&regionCode=KR&part=snippet,statistics&chart=mostPopular",
  "kind": "youtube#videoListResponse",
  "etag": "xs5iE2N49eLiB0YFPF7w3yb0d9s",
  "items": [
    {
      "kind": "youtube#video",
      "etag": "igBWdViroh70_QzDsTcs_eyJAYo",
      "id": "SRyk6t77Er8",
      "snippet": {
        "publishedAt": "2023-02-25T03:00:08Z",
        "channelId": "UCDNvRZRgvkBTUkQzFoT_8rA",
        "title": "[ENG] 커피 한잔은 핑계고",
        "description": "6년 전, 동네 주민인 태현이랑 커피 한잔 하자~ 했는데 그걸 지금에야 하네요?\n마침 연석이도 시간이 맞아 집 근처에서 모였는데요.\n\n사실 다~~ 핑계고🐓\n떠들어제끼는 걸로 멍뭉이판 한번 만들어 보려고 만났습니다🔥\n\nㅡㅡㅡ\n유선배 복지 콘텐츠 🐓핑계고🐓\n우리도 언제 올릴지 몰라요 ⚠불시 업로드⚠\n☁구독 & 좋아요 & 알림설정 필수☁\n\n👇’뜬뜬’ 소식 보러가기\nhttps://instagram.com/ddeun._.ddeun\n\n#뜬뜬 #핑계고 #유재석 #차태현 #유연석 #커피\n.\n.\n*본 콘텐츠와 무관한 내용의 댓글과 출연자에 대한 무분별한 욕설, 비난 및 조롱 등 악의적인 댓글은 무통보 삭제될 수 있습니다.",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/SRyk6t77Er8/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/SRyk6t77Er8/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/SRyk6t77Er8/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/SRyk6t77Er8/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/SRyk6t77Er8/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "뜬뜬 DdeunDdeun",
        "categoryId": "23",
        "liveBroadcastContent": "none",
        "localized": {
          "title": "[ENG] 커피 한잔은 핑계고",
          "description": "6년 전, 동네 주민인 태현이랑 커피 한잔 하자~ 했는데 그걸 지금에야 하네요?\n마침 연석이도 시간이 맞아 집 근처에서 모였는데요.\n\n사실 다~~ 핑계고🐓\n떠들어제끼는 걸로 멍뭉이판 한번 만들어 보려고 만났습니다🔥\n\nㅡㅡㅡ\n유선배 복지 콘텐츠 🐓핑계고🐓\n우리도 언제 올릴지 몰라요 ⚠불시 업로드⚠\n☁구독 & 좋아요 & 알림설정 필수☁\n\n👇’뜬뜬’ 소식 보러가기\nhttps://instagram.com/ddeun._.ddeun\n\n#뜬뜬 #핑계고 #유재석 #차태현 #유연석 #커피\n.\n.\n*본 콘텐츠와 무관한 내용의 댓글과 출연자에 대한 무분별한 욕설, 비난 및 조롱 등 악의적인 댓글은 무통보 삭제될 수 있습니다."
        },
        "defaultAudioLanguage": "ko"
      },
      "statistics": {
        "viewCount": "2914711",
        "likeCount": "69181",
        "favoriteCount": "0",
        "commentCount": "4837"
      }
    },
    {
      "kind": "youtube#video",
      "etag": "DL41DQhHjR2cQ5Z8_xQNY6whMp4",
      "id": "Uop1xS8EgUw",
      "snippet": {
        "publishedAt": "2023-02-27T08:00:21Z",
        "channelId": "UCyn-K7rZLXjGl7VXGweIlcA",
        "title": "[백종원 시장이 되다 7화] 여기에 목숨 걸 거 아니면 하지 마세요. 예산 시장엔 과연 무슨 일이...?",
        "description": "#백종원 #백종원시장이되다 #예산\n\n\n이제 시작이다!\n예산 시장 오픈을 앞두고 창업자 대 모집~!!\n\n하지만 쉽지 않을 거예요~\n사람이 많이 없어서 음식을 버릴 수도 있고\n음식 값이 싸서 힘들 수도 있고\n부담감으로 어려울 수도 있습니다...만!\n우리 같이 신나게 역사 한번 만들어 보자고요~!!\n\n예산 시장 D-day를 앞두고 \n더본 조리개발원과 창업자들의 피나는 연습과 노력!\n\n그런데!!\n갑자기 고성이 오가는 식당?!!\n과연, 이곳에 무슨 일이 일어난 걸까요?\n\n\n\n_\n00:00 인트로\n00:51 창업자 면접\n03:52 레시피 교육과 연습\n06:09 매장 깜짝 점검😨\n10:15 갈등...💦\n13:48 격려 및 위생등급제 동의!\n15:19 매장 재방문✨\n17:14 예고편!",
        "thumbnails": {
          "default": {
            "url": "https://i.ytimg.com/vi/Uop1xS8EgUw/default.jpg",
            "width": 120,
            "height": 90
          },
          "medium": {
            "url": "https://i.ytimg.com/vi/Uop1xS8EgUw/mqdefault.jpg",
            "width": 320,
            "height": 180
          },
          "high": {
            "url": "https://i.ytimg.com/vi/Uop1xS8EgUw/hqdefault.jpg",
            "width": 480,
            "height": 360
          },
          "standard": {
            "url": "https://i.ytimg.com/vi/Uop1xS8EgUw/sddefault.jpg",
            "width": 640,
            "height": 480
          },
          "maxres": {
            "url": "https://i.ytimg.com/vi/Uop1xS8EgUw/maxresdefault.jpg",
            "width": 1280,
            "height": 720
          }
        },
        "channelTitle": "백종원 PAIK JONG WON",
        "tags": [
          "백종원",
          "시장",
          "백종원예산",
          "찐맛집",
          "정보",
          "꿀팁",
          "창업",
          "재개발",
          "상권",
          "음식",
          "요리",
          "관광",
          "여행",
          "맛집",
          "레시피",
          "음식여행",
          "불맛",
          "정육점",
          "고기",
          "더본코리아",
          "paik jong won",
          "시장먹방",
          "얘산",
          "백종원시장",
          "백대표",
          "야시장",
          "비법",
          "창업교육",
          "골목식당",
          "탄생비화",
          "theborn"
        ],
        "categoryId": "22",
        "liveBroadcastContent": "none",
        "defaultLanguage": "ko",
        "localized": {
          "title": "[백종원 시장이 되다 7화] 여기에 목숨 걸 거 아니면 하지 마세요. 예산 시장엔 과연 무슨 일이...?",
          "description": "#백종원 #백종원시장이되다 #예산\n\n\n이제 시작이다!\n예산 시장 오픈을 앞두고 창업자 대 모집~!!\n\n하지만 쉽지 않을 거예요~\n사람이 많이 없어서 음식을 버릴 수도 있고\n음식 값이 싸서 힘들 수도 있고\n부담감으로 어려울 수도 있습니다...만!\n우리 같이 신나게 역사 한번 만들어 보자고요~!!\n\n예산 시장 D-day를 앞두고 \n더본 조리개발원과 창업자들의 피나는 연습과 노력!\n\n그런데!!\n갑자기 고성이 오가는 식당?!!\n과연, 이곳에 무슨 일이 일어난 걸까요?\n\n\n\n_\n00:00 인트로\n00:51 창업자 면접\n03:52 레시피 교육과 연습\n06:09 매장 깜짝 점검😨\n10:15 갈등...💦\n13:48 격려 및 위생등급제 동의!\n15:19 매장 재방문✨\n17:14 예고편!"
        },
        "defaultAudioLanguage": "ko"
      },
      "statistics": {
        "viewCount": "911697",
        "likeCount": "12917",
        "favoriteCount": "0",
        "commentCount": "5595"
      }
    }
  ],
  "nextPageToken": "CB4QAA",
  "pageInfo": {
    "totalResults": 200,
    "resultsPerPage": 30
  }
}
