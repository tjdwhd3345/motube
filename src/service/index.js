import Youtube from "./youtube";
import YoutubeClient from "./youtubeClient";

// eslint-disable-next-line no-undef
const youtubeClient = new YoutubeClient(process.env.REACT_APP_YOUTUBE_API_KEY);
export const youtube = new Youtube(youtubeClient); // 컴포넌트 내에서 과도한 비즈니스 코드를 방지하기 위함.
