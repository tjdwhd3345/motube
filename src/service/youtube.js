const { default: axios } = require('axios');

class Youtube {
  constructor(key) {
    this.youtube = axios.create({
      baseURL: 'https://youtube.googleapis.com/youtube/v3/',
      params: {
        key: key,
        maxResults: 3,
        regionCode: 'KR',
      },
    });
  }

  async mostPopular() {
    try {
      const res = await this.youtube.get('videos', {
        params: {
          part: 'snippet,statistics',
          chart: 'mostPopular',
        },
      });
      const results = res.data.items;
      console.log('mostPopular:', results);
      return results.map((result) => {
        return {
          id: result.id,
          snippet: result.snippet,
          channelId: result.snippet.channelId,
          statistics: result.statistics,
        };
      });
    } catch (err) {
      console.error('get mostPopular error:', err);
    }
  }

  async search(keyword) {
    try {
      const res = await this.youtube.get('search', {
        params: {
          part: 'snippet',
          type: 'video',
          q: keyword,
        },
      });
      const results = res.data.items;
      return results.map((result) => {
        return {
          id: result.id.videoId,
          snippet: result.snippet,
        };
      });
    } catch (err) {
      console.error('get search videos error:', err);
    }
  }

  async findVideo(videoId) {
    try {
      console.log('findVideo:', videoId);
      const res = await this.youtube.get('videos', {
        params: {
          part: 'snippet,statistics',
          id: videoId,
        },
      });
      const results = res.data.items;
      console.log('video, ', results);
      return results.map((result) => {
        return {
          id: result.id,
          snippet: result.snippet,
          channelId: result.snippet.channelId,
          statistics: result.statistics,
        };
      });
    } catch (err) {
      console.error('get video error:', err);
    }
  }

  async channels(videos) {
    try {
      const promises = videos.map(async (video) => {
        return await this.youtube.get('channels', {
          params: {
            part: 'snippet',
            id: video.channelId,
          },
        });
      });
      const results = await Promise.all(promises);
      console.log('results:', results);
      return results.map((result) => {
        return {
          thumbnails: result.data.items[0].snippet.thumbnails,
          statistics: result.data.items[0].statistics,
        };
      });
      /* for (let video of videos) {
        console.log('channels, ', video);
        const res = await this.youtube.get('channels', {
          params: {
            part: 'snippet,statistics',
            id: video.channelId,
          },
        });
        const results = res.data.items;
        console.log(results);
        return results.map((result) => {
          return {
            thumbnails: result.thumbnails,
            statistics: result.statistics,
          };
        });
      }
      return ''; */
    } catch (err) {
      // console.err('get channel info:', err);
    }
  }
}

export default Youtube;
