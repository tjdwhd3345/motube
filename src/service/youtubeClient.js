import axios from "axios";

class YoutubeClient {
  constructor(key) {
    this.youtube = axios.create({
      baseURL: "https://youtube.googleapis.com/youtube/v3/",
      params: {
        key: key,
        maxResults: 30,
        regionCode: "KR",
      },
    });
  }
  async videos(params) {
    return await this.youtube.get("videos", params);
  }
  async search(params) {
    return await this.youtube.get("search", params);
  }
  async channels(params) {
    return await this.youtube.get("channels", params);
  }
}

export default YoutubeClient;
