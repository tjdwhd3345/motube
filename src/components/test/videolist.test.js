import React from "react";
import { render, screen } from "@testing-library/react";
import { youtube } from "../../service";
import videos from "../../data/videos.json";
import channel1 from "../../data/channel1.json";
import channel2 from "../../data/channel2.json";
import VideoList from "../video_list/videolist";

const onVideoClick = jest.fn();
jest.mock("../../service");

const channels = [channel1, channel2];

describe("VideoList", () => {
  beforeEach(() => {
    youtube.mostPopular.mockImplementation(() => videos.items);
    youtube.channels.mockImplementation(() =>
      videos.items.map((item, index) => ({
        ...item,
        channelThumbnail: channels[index].items[0].snippet.thumbnails.default,
      }))
    );
  });
  afterEach(() => {
    youtube.mostPopular.mockReset();
    youtube.channels.mockReset();
  });

  it("renders correctly", async () => {
    const { asFragment } = render(<VideoList onVideoClick={onVideoClick} />);

    await screen.findAllByRole("listitem");
    expect(asFragment()).toMatchSnapshot();
  });
});
