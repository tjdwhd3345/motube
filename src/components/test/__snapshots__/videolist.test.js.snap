// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList renders correctly 1`] = `
<DocumentFragment>
  <ul
    class="videos"
  >
    <li
      class="container grid"
    >
      <div
        class="video"
      >
        <img
          alt="video thumbnail"
          class="thumbnail"
          src="https://i.ytimg.com/vi/SRyk6t77Er8/maxresdefault.jpg"
        />
        <div
          class="metadata"
        >
          <img
            alt="channelThumbnail"
            class="channelThumbnail"
            src="https://yt3.ggpht.com/aOgjye3sMIxNl2SW2wAQZpZWUXzZ5Rg0rNITacRQKVfXvF9cnPWb77G3_gH5s2Zyw241BXWYWg=s88-c-k-c0x00ffffff-no-rj"
          />
          <div
            class="titlebox"
          >
            <span
              class="title"
            >
              [ENG] 커피 한잔은 핑계고
            </span>
            <span
              class="channel"
            >
              뜬뜬 DdeunDdeun
            </span>
            <span
              class="channelInfo"
            >
              조회수 2,914,711 회 • 2023.02.25
            </span>
          </div>
        </div>
      </div>
    </li>
    <li
      class="container grid"
    >
      <div
        class="video"
      >
        <img
          alt="video thumbnail"
          class="thumbnail"
          src="https://i.ytimg.com/vi/Uop1xS8EgUw/maxresdefault.jpg"
        />
        <div
          class="metadata"
        >
          <img
            alt="channelThumbnail"
            class="channelThumbnail"
            src="https://yt3.ggpht.com/J3OI66Bc7T3nheyKJKAkXR6tb-_bpCsoRMTFoslOBCXI3DpVY8eFY4LZWzww3BEgkRjMOEoI=s88-c-k-c0x00ffffff-no-rj"
          />
          <div
            class="titlebox"
          >
            <span
              class="title"
            >
              [백종원 시장이 되다 7화] 여기에 목숨 걸 거 아니면 하지 마세요. 예산 시장엔 과연 무슨 일이...?
            </span>
            <span
              class="channel"
            >
              백종원 PAIK JONG WON
            </span>
            <span
              class="channelInfo"
            >
              조회수 911,697 회 • 2023.02.27
            </span>
          </div>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
